tests/integration/olp-cpp-sdk-dataservice-read/VolatileLayerClientTest.cpp

```diff
@@ -923,4 +971,407 @@ TEST_F(DataserviceReadVolatileLayerClientTest, RemoveFromCacheTileKey) {
   ASSERT_FALSE(data_response.IsSuccessful())
       << ApiErrorToString(data_response.GetError());
 }
+
+TEST_F(DataserviceReadVolatileLayerClientTest, PrefetchTilesWithCache) {
+  olp::client::HRN catalog(GetTestCatalog());
+  constexpr auto kLayerId = "hype-test-prefetch";
+  olp::dataservice::read::VolatileLayerClient client(catalog, kLayerId,
+                                                     settings_);
+  {
+    SCOPED_TRACE("Prefetch tiles online and store them in memory cache");
+
+    std::vector<olp::geo::TileKey> tile_keys = {
+        olp::geo::TileKey::FromHereTile("5904591")};
+
+    auto request = olp::dataservice::read::PrefetchTilesRequest()
+                       .WithTileKeys(tile_keys)
+                       .WithMinLevel(11)
+                       .WithMaxLevel(12);
+
+    auto promise = std::make_shared<std::promise<PrefetchTilesResponse>>();
+    std::future<PrefetchTilesResponse> future = promise->get_future();
+    auto token = client.PrefetchTiles(
+        request, [promise](PrefetchTilesResponse response) {
+          promise->set_value(std::move(response));
+        });
+
+    ASSERT_NE(future.wait_for(kTimeout), std::future_status::timeout);
+    PrefetchTilesResponse response = future.get();
+    ASSERT_TRUE(response.IsSuccessful()) << response.GetError().GetMessage();
+    ASSERT_FALSE(response.GetResult().empty());
+
+    const auto& result = response.GetResult();
+
+    for (auto tile_result : result) {
+      ASSERT_TRUE(tile_result->IsSuccessful());
+      ASSERT_TRUE(tile_result->tile_key_.IsValid());
+    }
+  }
+
+  {
+    SCOPED_TRACE("Read cached data from pre-fetched sub-partition #1");
+    auto promise = std::make_shared<std::promise<DataResponse>>();
+    std::future<DataResponse> future = promise->get_future();
+    auto token = client.GetData(olp::dataservice::read::DataRequest()
```

Comment:
Please follow the same approach as above. The request is separated, the checks are separated from the request.